src/nimblepkg/forge_aliases.nim

```diff
@@ -0,0 +1,120 @@
+# Copyright (C) Dominik Picheta. All rights reserved.
+# BSD License. Look at license.txt for more info.
+
+import std/strutils
+import common
+
+type
+  ForgeKind* = enum
+    fgGitHub
+    fgGitLab
+    fgSourceHut
+    fgCodeberg
+
+  Forge* = ref object
+    case kind*: ForgeKind
+    of fgGitHub:
+      ghUsername*, ghRepo*: string
+    of fgGitLab:
+      glUsername*, glRepo*: string
+    of fgSourceHut:
+      shUsername*, shRepo*: string
+    of fgCodeberg:
+    cbUsername*, cbRepo*: string
+
+proc expand*(alias: Forge): string {.inline.} =
+  var expanded = "https://" # add an option to use http instead?
+
+  case alias.kind
+  of fgGitHub:
+    expanded &= "github.com/" & alias.ghUsername & '/' & alias.ghRepo
+  of fgGitLab:
+    expanded &= "gitlab.com/" & alias.glUsername & '/' & alias.glRepo
+  of fgSourceHut:
+    expanded &= "git.sr.ht/" & alias.shUsername & '/' & alias.shRepo
+  of fgCodeberg:
+    expanded &= "codeberg.org/" & alias.cbRepo & '/' & alias.cbRepo
+
+  expanded
+
+proc isForgeAlias*(value: string): bool {.inline.} =
+  let splitted = value.split(':')
+
+  splitted.len == 2
+
+proc parseForgeKind*(value: string): ForgeKind {.inline.} =
+  let splitted = value.split(':')
+  
+  case splitted[0].toLowerAscii()
+  of "github", "gh": return fgGitHub
+  of "gitlab", "gl": return fgGitLab
+  of "sourcehut", "srht", "shart": return fgSourceHut
+  of "codeberg", "cb", "cberg": return fgCodeberg
```

Comment:
Codeberg is one of many [forgejo](https://forgejo.org/) instances, perhaps extract Codeberg's logic out into a new "forgejo" type, allowing e.g. `forgejo:git.example.org:foo/bar`. It would most likely work for legacy Gitea instances too. You could do the same for sourcehut but selfhosted sourcehut instances are much rarer.

Reply:
Yeah, but these are for forge providers, not forge services. You might as well put in the full URL at that point.